src/serac/numerics/functional/detail/hexahedron_H1.inl

```diff
@@ -204,85 +228,195 @@ struct finite_element<mfem::Geometry::CUBE, H1<p, c> > {
       gradient(i, 1) = contract<0, 1>(A22, B);
       gradient(i, 2) = contract<0, 1>(A20, G);
     }
+#else
+
+    RAJA_TEAM_SHARED tensor<double, c, q, q, q> value;
+    RAJA_TEAM_SHARED tensor<double, c, dim, q, q, q> gradient;
+    constexpr auto                                   B = calculate_B<apply_weights, q>();
+    constexpr auto                                   G = calculate_G<apply_weights, q>();
+    for (int i = 0; i < c; i++) {
+      RAJA_TEAM_SHARED decltype(deduce_contract_return_type<2, 1>(X[i], B)) A10;
+      RAJA_TEAM_SHARED decltype(deduce_contract_return_type<2, 1>(X[i], G)) A11;
+      RAJA_TEAM_SHARED decltype(deduce_contract_return_type<1, 1>(A10, B))  A20;
+      RAJA_TEAM_SHARED decltype(deduce_contract_return_type<1, 1>(A11, B))  A21;
+      RAJA_TEAM_SHARED decltype(deduce_contract_return_type<1, 1>(A10, G))  A22;

```

Comment:
This is considerably more shared memory than is needed. 
Since these arrays are not all in use at the same time, we can get away with a smaller allocation. See the snippets in https://github.com/LLNL/serac/files/12482282/mfem_dynamic_kernel_refactor.pdf section "An alternative implementation" for details.